shimCPURepro/backend_mismatch_reproducer.cpp

```diff
@@ -0,0 +1,257 @@
+#include <hip/hip_runtime.h>
+#include <hip/hip_interop.h>
+#include <iostream>
+#include <vector>
+#include <cmath>
+#include <cstring>
+
+#define HIP_CHECK(status) \
+    if (status != hipSuccess) { \
+        std::cerr << "HIP error: " << hipGetErrorString(status) << " at line " << __LINE__ << std::endl; \
+        exit(1); \
+    }
+
+// For hipGetBackendNativeHandles which returns int, not hipError_t
+#define INTEROP_CHECK(status) \
+    if (status != 0) { \
+        std::cerr << "Interop error: " << status << " at line " << __LINE__ << std::endl; \
+        exit(1); \
+    }
+
+// Handle indices for debugging
+const char* HandleNames[] = {
+    "BACKEND_NAME",
+    "PLATFORM_DRIVER", 
+    "DEVICE", 
+    "CONTEXT", 
+    "QUEUE", 
+    "COMMAND_LIST"
+};
+
+// More detailed handle printing with names
+void print_handles(const char* prefix, const std::vector<uintptr_t>& handles) {
+    std::cout << "\n=== " << prefix << " (" << handles.size() << " handles) ===" << std::endl;
+    
+    for (size_t i = 0; i < handles.size(); i++) {
+        const char* name = (i < sizeof(HandleNames)/sizeof(HandleNames[0])) ? 
```

Comment:
"sizeof(HandleNames)/sizeof(HandleNames[0])" looks to me a constant value. Why can't you calculate it before hand.